src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -612,4 +640,82 @@ std::string primitive_inst::generic_to_string(program_node const& node, const ch
 
     return primitive_description.str();
 }
+
+cldnn::network::ptr primitive_inst::get_unfused_subgraph() {
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+    GPU_DEBUG_IF(debug_config->verbose >= 4) {
+        GPU_DEBUG_COUT << id() << ": Use unfused subgraph due to unexpected fusions\n";
+    }
+    if (!_unfused_subgraph) {
+        topology t;
+
+        // Add input primitives: constants are moved as is
+        // Any other primitive types are replaced with input_layout
+        for (auto& dep : _node.get_dependencies()) {
+            if (dep->is_type<data>()) {
+                t.add(*dep->as<data>().get_primitive());
+            } else {
+                input_layout in_prim(dep->id(), dep->get_output_layout());
+                t.add(in_prim);
+            }
+        }
+
+        // Create the primitive itself
+        t.add_primitive(std::const_pointer_cast<primitive>(_node.get_primitive()));
+
+        // Add primitives for fused-ops
+        for (auto& fd : _impl_params->fused_desc) {
+            t.add_primitive(std::const_pointer_cast<primitive>(fd.desc));
+        }
+
+        build_options bo;
+        bo.set_option(build_option::allow_static_input_reorder(true));
+        auto prog = program::build_program(get_network().get_engine(), t, bo, true, false);
+
+        _unfused_subgraph = network::allocate_network(get_network().get_stream_ptr(), prog, true, get_network().is_primary_stream());
+    }
+    return _unfused_subgraph;
+}
+
+bool primitive_inst::is_valid_fusion() const {
+    if (!is_dynamic())
+        return true;
+
+    auto fuse_descriptors = _impl_params->fused_desc;
+    if (fuse_descriptors.empty())
+        return true;
+
+    std::vector<fused_primitive_desc> fused_eltwise_prims;
+    for (auto& fd : fuse_descriptors) {
+        if (fd.is_type<eltwise>()) {
+            fused_eltwise_prims.push_back(fd);
+        }
+    }
+
+    if (fused_eltwise_prims.empty())
+        return true;
+
+    auto out_pshape = _impl_params->output_layout.get_partial_shape();
+    for (auto& fd : fused_eltwise_prims) {
+        auto dep_idx = fd.dep_start_idx;
+        OPENVINO_ASSERT(fd.total_num_deps == 2, "[GPU] Unexpected count of dependencies in dynamic fusion for eltwise");
+        OPENVINO_ASSERT(_deps.size() >= dep_idx, "[GPU] Invalid fused dependency idx");
+        auto dep = _deps[dep_idx];
+
+        if (dep->get_node().is_type<data>())
+            continue;
```

Comment:
Can we ensure that it is always valid if the additional dep is data? 
For example, what if the fused_nodes' input is smaller than const input?

Reply:
Good point. Will remove this check to have shape comparison working for data deps as well.

Reply:
Updated